pkg/controller/hibernation/aws_actuator.go

```diff
@@ -0,0 +1,148 @@
+package hibernation
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	log "github.com/sirupsen/logrus"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1"
+	awsclient "github.com/openshift/hive/pkg/awsclient"
+)
+
+func init() {
+	RegisterActuator(&awsActuator{awsClientFn: getAWSClient})
+}
+
+type awsActuator struct {
+	// awsClientFn is the function to build an AWS client, here for testing
+	awsClientFn func(log.FieldLogger, *hivev1.ClusterDeployment, client.Client) (awsclient.Client, error)
+}
+
+// CanHandle returns true if the actuator can handle a particular ClusterDeployment
+func (a *awsActuator) CanHandle(cd *hivev1.ClusterDeployment) bool {
+	return cd.Spec.Platform.AWS != nil
+}
+
+// StopMachines will stop machines belonging to the given ClusterDeployment
+func (a *awsActuator) StopMachines(logger log.FieldLogger, cd *hivev1.ClusterDeployment, c client.Client) error {
+	logger = logger.WithField("cloud", "aws")
+	logger.Info("stopping machines")
+	awsClient, err := a.awsClientFn(logger, cd, c)
+	if err != nil {
+		return err
+	}
+	instanceIDs, err := getClusterInstanceIDs(logger, cd, awsClient, []string{"pending", "running"})
```

Comment:
How would you feel about some package-scoped variables like the following?
```
var (
	runningStates           = sets.NewString("running")
	stoppedStates           = sets.NewString("stopped")
	pendingStates           = sets.NewString("pending")
	stoppingStates          = sets.NewString("stopping", "shutting-down")
	runningOrPendingStates  = runningStates.Union(pendingStates)
	stoppedOrStoppingStates = stoppingStates.Union(stoppingStates)
	notRunningStates        = stoppedOrStoppingStates.Union(pendingStates)
	notStoppedStates        = runningOrPendingStates.Union(stoppingStates)
)
```

```suggestion
	instanceIDs, err := getClusterInstanceIDs(logger, cd, awsClient, runningOrPendingStates.List())
```